Yafc/Windows/PreferencesScreen.cs

```diff
@@ -81,6 +89,29 @@ public override void Build(ImGui gui) {
                 }
             }
 
+            IEnumerable<Technology> productivityTech = Database.technologies.all
+                .Where(x => x.changeRecipeProductivity.Count != 0)
+                .OrderBy(x => x.locName);
+            foreach (var tech in productivityTech) {
+                using (gui.EnterRow()) {
+                    var color = tech.IsAccessible() ? SchemeColor.Source : SchemeColor.SourceFaint;
+                    if (IconDisplayStyle.Default.UseScaleSetting) {
+                        Rect rect = gui.AllocateRect(IconDisplayStyle.Default.Size, IconDisplayStyle.Default.Size, RectAlignment.Middle);
+                        gui.DrawIcon(rect.Expand(IconDisplayStyle.Default.Size * (Project.current.preferences.iconScale - 1) / 2), tech.icon, color);
+                    }
+                    else {
+                        gui.BuildIcon(tech.icon, IconDisplayStyle.Default.Size, color);
+                    }
```

Comment:
This looks like you copied part of `BuildFactorioObjectIcon` instead of calling it. You could also call `BuildFactorioObjectButton` instead, which will let people middle-click the icon will open the dependency explorer.
```suggestion
                    gui.BuildFactorioObjectButton(tech , ButtonDisplayStyle.Default);
```
If you copied the method to remove the milestone display, you could pass `IconDisplayStyle.Default with { MilestoneDisplay = MilestoneDisplay.None }`. That said, I think the milestone display should be present; it gives users a hint that they probably shouldn't set that research level yet, since it's still locked.

Reply:
I agree that having the milestone there is nicer, not sure where I copied that code from, thanks!